pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -91,60 +80,60 @@ func syncCluster(
 				if wait < maxWait {
 					wait = wait * 2
 				}
-				shared.LogWarnf(
+				shared.LogErrorf(
+					reqLogger,
+					updateErr,
 					cr,
 					shared.EventReasonCluster,
-					"trying status update again in %v; failed because: %v",
+					"trying status update again in %v; failed",
 					wait,
-					updateErr,
 				)
 				time.Sleep(wait)
 			}
 		}
 	}()
 
-	// Ignore stale poll-driven handler for a resource we have since
-	// updated. Also for a new CR just update the status state/gen.
-	shouldProcessCR := handleStatusGen(cr, handler)
-
 	// Regardless of whether the status gen is as expected, make sure the CR
```

Comment:
Should remove the "regardless" clause here.

Reply:
-       // Regardless of whether the status gen is as expected, make sure the CR
-       // finalizers are as we want them. We use a finalizer to prevent races
-       // between polled CR updates and CR deletion.
+       // We use a finalizer to ensure that only KubeDirector updates status

Reply:
Not sure what happened there. The new comment reads, "We use a finalizer to ensure that only KubeDirector updates status"